app/lib/features/pins/pages/create_pin_page.dart

```diff
@@ -0,0 +1,353 @@
+        child: Form(
+          key: _formKey,
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              const SizedBox(height: 15),
+              Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  _buildTitleField(),
+                  const SizedBox(height: 15),
+                  _buildLinkField(),
+                  const SizedBox(height: 15),
+                  _buildAttachmentField(),
+                  const SizedBox(height: 15),
+                  if (attachments.isNotEmpty)
+                    Wrap(
+                      spacing: 5.0,
+                      runSpacing: 10.0,
+                      children: <Widget>[
+                        for (var pinAttachment in attachments)
+                          _AttachmentItemWidget(pinAttachment),
+                      ],
+                    ),
+                  const SizedBox(height: 15),
+                  _buildDescriptionField(),
+                ],
+              ),
+              const SizedBox(height: 15),
+              const SelectSpaceFormField(canCheck: 'CanPostPin'),
+            ],
+          ),
+        ),
+      ),
+      confirmActionTitle: 'Create Pin',
+      cancelActionTitle: null,
+      confirmActionOnPressed: () async {
+        if (_formKey.currentState!.validate()) {
+          _handleCreatePin();
+        }
+      },
+    );
+  }
+
+  Widget _buildTitleField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Title'),
+        ),
+        InputTextField(
+          hintText: 'Pin Name',
+          key: CreatePinPage.titleFieldKey,
+          textInputType: TextInputType.text,
+          controller: _titleController,
+          validator: (value) => (value != null && value.trim().isNotEmpty)
+              ? null
+              : 'Please enter a title',
+        ),
+      ],
+    );
+  }
+
+  Widget _buildAttachmentField() {
+    return InkWell(
+      onTap: () => PinUtils.showAttachmentSelection(context, ref),
+      child: Container(
+        height: 100,
+        width: double.infinity,
+        decoration: BoxDecoration(
+          color: Theme.of(context).colorScheme.primaryContainer,
+          border: Border.all(
+            color: Theme.of(context).colorScheme.primary,
+            strokeAlign: BorderSide.strokeAlignOutside,
+          ),
+          borderRadius: BorderRadius.circular(12),
+        ),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Icon(
+              Atlas.file_arrow_up_thin,
+              size: 14,
+              color: Theme.of(context).colorScheme.neutral5,
+            ),
+            Text(
+              'Upload File',
```

Comment:
```suggestion
              'Upload Attachment',
```
As `file` is a specific type of attachment, no?